PATCH network peering failed with no peer found

The initial deployment succeeded with peer successfully created, but subsequent deployment failed with following error even without any changes. The error indicates that the peer is Not Found under the project, but the peer does exist.

module.dev_jcdigital_test345_prep.null_resource.prep (local-exec): WARNING: The behavior of this command has been altered by the following extension: aks-preview
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: 2019/08/30 11:15:50 [DEBUG] MongoDB Atlas API Response Details:
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: ---[ RESPONSE ]--------------------------------------
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: HTTP/2.0 404 Not Found
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: Content-Length: 228
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: Content-Type: application/json
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: Date: Fri, 30 Aug 2019 18:15:50 GMT
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: Strict-Transport-Security: max-age=300
2019-08-30T11:15:50.785-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: {
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:  "detail": "No peer with ID 5d696758014b767112cf712c exists in project 5d68017b014b767112c77330.",
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:  "error": 404,
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:  "errorCode": "PEER_NOT_FOUND",
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:  "parameters": [
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:   "5d696758014b767112cf712c",
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:   "5d68017b014b767112c77330"
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:  ],
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas:  "reason": "Not Found"
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: }
2019-08-30T11:15:50.786-0700 [DEBUG] plugin.terraform-provider-mongodbatlas: -----------------------------------------------------
2019/08/30 11:15:50 [DEBUG] module.dev_jcdigital_test345_atlas.mongodbatlas_network_peering.rest: apply errored, but we're indicating that via the Error pointer rather than returning it: error updating MongoDB Network Peering Connection (5d696758014b767112cf712c): PATCH https://cloud.mongodb.com/api/atlas/v1.0/groups/5d68017b014b767112c77330/peers/5d696758014b767112cf712c: 404 (request "Not Found") No peer with ID 5d696758014b767112cf712c exists in project 5d68017b014b767112c77330.
2019/08/30 11:15:50 [ERROR] module.dev_jcdigital_test345_atlas: eval: *terraform.EvalApplyPost, err: error updating MongoDB Network Peering Connection (5d696758014b767112cf712c): PATCH https://cloud.mongodb.com/api/atlas/v1.0/groups/5d68017b014b767112c77330/peers/5d696758014b767112cf712c: 404 (request "Not Found") No peer with ID 5d696758014b767112cf712c exists in project 5d68017b014b767112c77330.
2019/08/30 11:15:50 [ERROR] module.dev_jcdigital_test345_atlas: eval: *terraform.EvalSequence, err: error updating MongoDB Network Peering Connection (5d696758014b767112cf712c): PATCH https://cloud.mongodb.com/api/atlas/v1.0/groups/5d68017b014b767112c77330/peers/5d696758014b767112cf712c: 404 (request "Not Found") No peer with ID 5d696758014b767112cf712c exists in project 5d68017b014b767112c77330.

cdsfmbbl1:bob-jc-sense-tf cdsfblo$  curl --user "unllvfyp:<redacted>" --digest \
>   --header "Accept: application/json" \
>    -- request GET "https://cloud.mongodb.com/api/atlas/v1.0/groups/5d68017b014b767112c77330/peers/5d696758014b767112cf712c"
curl: (6) Could not resolve host: request
curl: (6) Could not resolve host: GET
{"azureDirectoryId":"bd2ac3b4-122a-4000-ac98-779c8efd4722","azureSubscriptionId":"38171a42-9a9e-4fa6-ac1f-43b126efd341","containerId":"5d68017d79358ee73ce91501","errorState":null,"id":"5d696758014b767112cf712c","resourceGroupName":"dev-jcdigital-test345-rg","status":"AVAILABLE","vnetName":"dev-jcdigital-test345-vnet"}

--- a/mongodbatlas/resource_mongodbatlas_network_peering.go
+++ b/mongodbatlas/resource_mongodbatlas_network_peering.go
@@ -95,6 +95,7 @@ func resourceMongoDBAtlasNetworkPeering() *schema.Resource {
 			"atlas_cidr_block": {
 				Type:     schema.TypeString,
 				Optional: true,
+				ForceNew: true,
 				Computed: true,
 			},
 			"azure_directory_id": {
@@ -305,9 +306,6 @@ func resourceMongoDBAtlasNetworkPeeringRead(d *schema.ResourceData, meta interfa
 	if err := d.Set("status_name", peer.StatusName); err != nil {
 		return fmt.Errorf("error setting `status_name` for Network Peering Connection (%s): %s", peerID, err)
 	}
-	if err := d.Set("atlas_cidr_block", peer.AtlasCIDRBlock); err != nil {
-		return fmt.Errorf("error setting `atlas_cidr_block` for Network Peering Connection (%s): %s", peerID, err)
-	}
 	if err := d.Set("azure_directory_id", peer.AzureDirectoryID); err != nil {
 		return fmt.Errorf("error setting `azure_directory_id` for Network Peering Connection (%s): %s", peerID, err)
 	}
@@ -369,10 +367,6 @@ func resourceMongoDBAtlasNetworkPeeringUpdate(d *schema.ResourceData, meta inter
 
 	if d.HasChange("vpc_id") {
 		peer.VpcID = d.Get("vpc_id").(string)
-	}
-
-	if d.HasChange("atlas_cidr_block") {
-		peer.AtlasCIDRBlock = d.Get("atlas_cidr_block").(string)
 	}
 
 	if d.HasChange("azure_directory_id") {

--- a/mongodbatlas/resource_mongodbatlas_network_peering_test.go
+++ b/mongodbatlas/resource_mongodbatlas_network_peering_test.go
@@ -13,7 +13,6 @@ import (
 )
 
 func TestAccResourceMongoDBAtlasNetworkPeering_basicAWS(t *testing.T) {
-
 	var peer matlas.Peer
 
 	resourceName := "mongodbatlas_network_peering.test"
@@ -33,7 +32,7 @@ func TestAccResourceMongoDBAtlasNetworkPeering_basicAWS(t *testing.T) {
 				Config: testAccMongoDBAtlasNetworkPeeringConfigAWS(projectID, providerName, vpcID, awsAccountID, vpcCIDRBlock, awsRegion),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckMongoDBAtlasNetworkPeeringExists(resourceName, &peer),
-					testAccCheckMongoDBAtlasNetworkPeeringAttributes(&peer, vpcCIDRBlock),
+					testAccCheckMongoDBAtlasNetworkPeeringAttributes(&peer, providerName),
 					resource.TestCheckResourceAttrSet(resourceName, "project_id"),
 					resource.TestCheckResourceAttrSet(resourceName, "container_id"),
 					resource.TestCheckResourceAttr(resourceName, "provider_name", providerName),
@@ -54,7 +53,6 @@ func TestAccResourceMongoDBAtlasNetworkPeering_basicAWS(t *testing.T) {
 }
 
 func TestAccResourceMongoDBAtlasNetworkPeering_basicAzure(t *testing.T) {
-	t.Skip()
 	var peer matlas.Peer
 
 	resourceName := "mongodbatlas_network_peering.test"
@@ -71,7 +69,7 @@ func TestAccResourceMongoDBAtlasNetworkPeering_basicAzure(t *testing.T) {
 		CheckDestroy: testAccCheckMongoDBAtlasNetworkPeeringDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccMongoDBAtlasNetworkPeeringConfigAzure(projectID, providerName, directoryID, subcrptionID, vNetName, resourceGroupName),
+				Config: testAccMongoDBAtlasNetworkPeeringConfigAzure(projectID, providerName, directoryID, subcrptionID, resourceGroupName, vNetName),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckMongoDBAtlasNetworkPeeringExists(resourceName, &peer),
 					testAccCheckMongoDBAtlasNetworkPeeringAttributes(&peer, providerName),
@@ -87,7 +85,7 @@ func TestAccResourceMongoDBAtlasNetworkPeering_basicAzure(t *testing.T) {
 				ImportStateIdFunc:       testAccCheckMongoDBAtlasNetworkPeeringImportStateIDFunc(resourceName),
 				ImportState:             true,
 				ImportStateVerify:       true,
-				ImportStateVerifyIgnore: []string{"region_name"},
+				ImportStateVerifyIgnore: []string{"region_name", "atlas_cidr_block"},
 			},
 		},
 	})
@@ -156,6 +154,7 @@ func testAccCheckMongoDBAtlasNetworkPeeringExists(resourceName string, peer *mat
 
 		if peerResp, _, err := conn.Peers.Get(context.Background(), rs.Primary.Attributes["project_id"], rs.Primary.Attributes["peer_id"]); err == nil {
 			*peer = *peerResp
+			peer.ProviderName = rs.Primary.Attributes["provider_name"]
 			return nil
 		}
 
@@ -211,7 +210,7 @@ func testAccMongoDBAtlasNetworkPeeringConfigAWS(projectID, providerName, vpcID, 
 	`, projectID, providerName, vpcID, awsAccountID, vpcCIDRBlock, awsRegion)
 }
 
-func testAccMongoDBAtlasNetworkPeeringConfigAzure(projectID, providerName, directoryID, subcrptionID, vNetName, resourceGroupName string) string {
+func testAccMongoDBAtlasNetworkPeeringConfigAzure(projectID, providerName, directoryID, subcrptionID, resourceGroupName, vNetName string) string {
 	return fmt.Sprintf(`
 		resource "mongodbatlas_network_container" "test" {
 			project_id   		  = "%[1]s"
@@ -230,7 +229,7 @@ func testAccMongoDBAtlasNetworkPeeringConfigAzure(projectID, providerName, direc
 			resource_group_name   = "%[5]s"
 			vnet_name	            = "%[6]s"
 		}
-	`, projectID, providerName, directoryID, subcrptionID, vNetName, resourceGroupName)
+	`, projectID, providerName, directoryID, subcrptionID, resourceGroupName, vNetName)
 }
 
 func testAccMongoDBAtlasNetworkPeeringConfigGCP(projectID, providerName, gcpProjectID string) string {

